gtsam/hybrid/HybridFactorGraph.h

```diff
@@ -315,10 +315,12 @@ class HybridFactorGraph : protected FactorGraph<Factor>,
   /// The total number of factors in the Gaussian factor graph.
   DiscreteKeys discreteKeys() const {
     DiscreteKeys result;
-    // TODO(Frank): implement!
-    // DiscreteKeys result = discreteGraph_.discreteKeys();
-    // DiscreteKeys dcKeys = dcGraph_.discreteKeys();
-    // result += dcKeys
+    for(auto&& key: discreteGraph_.keys()) {
+      //TODO(Varun) how to get cardinality?
+      result.emplace_back(key, 2);
```

Comment:
This is the place that needs the most thought. I guess we'll have to record the DiscreteKeys in the discrete factors. Currently they only record the keys and discard the cardinalities.